spirv_msl.cpp

```diff
@@ -3533,19 +3824,87 @@ void CompilerMSL::emit_custom_functions()
 			statement("template<typename T> struct spvRemoveReference { typedef T type; };");
 			statement("template<typename T> struct spvRemoveReference<thread T&> { typedef T type; };");
 			statement("template<typename T> struct spvRemoveReference<thread T&&> { typedef T type; };");
-			statement("template<typename T> inline constexpr thread T&& spvForward(thread typename "
-			          "spvRemoveReference<T>::type& x)");
+
+			statement("template<typename T>");
+			statement(force_inline);
+			statement("constexpr thread T&& spvForward(thread typename spvRemoveReference<T>::type& x)");
 			begin_scope();
 			statement("return static_cast<thread T&&>(x);");
 			end_scope();
-			statement("template<typename T> inline constexpr thread T&& spvForward(thread typename "
-			          "spvRemoveReference<T>::type&& x)");
+
+			statement("template<typename T>");
+			statement(force_inline);
+			statement("constexpr thread T&& spvForward(thread typename spvRemoveReference<T>::type&& x)");
 			begin_scope();
 			statement("return static_cast<thread T&&>(x);");
 			end_scope();
 			statement("");
 			break;
 
+		case SPVFuncImplRowMajor2x3:
```

Comment:
This was removed in the grand buffer packing refactor. There should be no reason to reintroduce it.

Reply:
You're right, this is my fault. I didn't remove it properly when I rebased the changes from an older revision into a more recent one. It didn't break anything but indeed seems to be unnecessary now. I'll remove that.

Reply:
This is also removed in the first update.